### PR TITLE
doc: zbus: fix msg subscriber thread signatures

### PR DESCRIPTION
**Pull request by bbilas**
Fix the thread function signatures to match k_thread_entry_t.

### CODE REVIEW

**Review by kartben — APPROVED** [review]

File: doc/services/zbus/index.rst [hunk 1]
```diff
@@ -424,8 +424,11 @@ exchanges accelerometer data, for example.
     K_THREAD_DEFINE(subscriber_task_id, 512, subscriber_task, NULL, NULL, NULL, 3, 0, 0);
 
     ZBUS_MSG_SUBSCRIBER_DEFINE(my_msg_subscriber);
-    static void msg_subscriber_task(void *sub)
+    static void msg_subscriber_task(void *ptr1, void *ptr2, void *ptr3)
```
Comment (rodrigopex) [inline doc/services/zbus/index.rst:427]: This is not necessary if you are not using the arguments. The current signature is wrong indeed, but for solving that you need to remove the `sub` parameter. 
```suggestion
    static void msg_subscriber_task()
```
That should be enough.

File: doc/services/zbus/index.rst [hunk 2]
```diff
@@ -424,8 +424,11 @@ exchanges accelerometer data, for example.
...
     {
+            ARG_UNUSED(ptr1);
+            ARG_UNUSED(ptr2);
+            ARG_UNUSED(ptr3);
```
Comment (rodrigopex) [inline doc/services/zbus/index.rst:431]:
```suggestion

```

**Review by rodrigopex — CHANGES_REQUESTED** [review]
You can only remove the `sub` parameter from the signature.

File: doc/services/zbus/index.rst (hunk 1 above)
Comment (bbilas) [reply doc/services/zbus/index.rst:427]: The most common pattern in Zephyr that I saw is defined function with arguments even though they are unused. The same is done in the threads docs area, please see:

<pre>
#define MY_STACK_SIZE 500
#define MY_PRIORITY 5

extern void my_entry_point(void *, void *, void *);

K_THREAD_STACK_DEFINE(my_stack_area, MY_STACK_SIZE);
struct k_thread my_thread_data;

k_tid_t my_tid = k_thread_create(&my_thread_data, my_stack_area,
                                 K_THREAD_STACK_SIZEOF(my_stack_area),
                                 my_entry_point,
                                 NULL, NULL, NULL,
                                 MY_PRIORITY, 0, K_NO_WAIT);
</pre>

Comment (rodrigopex) [reply doc/services/zbus/index.rst:427]: A typical pattern if you take a look at the samples is using only `extern void my_entry_point(void)`. I cannot see any benefit in keeping the parameter there.

Comment (bbilas) [reply doc/services/zbus/index.rst:427]: Alright, I will change it.

Comment (kartben) [reply doc/services/zbus/index.rst:427]: > A typical pattern if you take a look at the samples is using only `extern void my_entry_point(void)`. I cannot see any benefit in keeping the parameter there.

please see https://github.com/zephyrproject-rtos/zephyr/pull/62903
Should the tasks have the wrong signature and inadvertently return, they will corrupt the stack. I think it's good practice to make sure the signature is correct, even if that causes extra `ARG_UNUSED` stuff...

Comment (rodrigopex) [reply doc/services/zbus/index.rst:427]: Interesting, so keep that @bbilas

Comment (bbilas) [reply doc/services/zbus/index.rst:427]: I had my finger on the trigger but you were faster ;)

Comment (rodrigopex) [reply doc/services/zbus/index.rst:427]: Based on that, we have other samples and test code to fix. :sweat_smile:

**Review by rodrigopex — APPROVED** [review]

**Pull request merged**